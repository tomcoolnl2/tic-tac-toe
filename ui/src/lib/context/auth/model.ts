import { User } from '@tic-tac-toe/model';
import { AuthError } from './error';

export interface AuthContextState {
	fetchUserName: () => Promise<string | AuthError | null>;
	login: (pwd: string) => Promise<AuthError | User>;
	logout: () => Promise<User>;
}
