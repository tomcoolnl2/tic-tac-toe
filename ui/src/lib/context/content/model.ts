import { Locale } from '@tic-tac-toe/model';

export type ContentAction =
	| { type: 'FETCH_START' }
	| { type: 'FETCH_SUCCESS'; payload: AppContent }
	| { type: 'FETCH_ERROR'; payload: Error }
	| { type: 'SET_LANGUAGE'; payload: Locale };

export interface ContentState {
	appContent: AppContent | null;
	isContentLoading: boolean;
	contentError: Error | null;
	locale: Locale;
}

export type ContentStateWithLanguageSelector = ContentState & {
	setLanguage: (locale: Locale) => void;
};

export interface AppContent {
	appTitle: string;
	loginScreen: AppScreenContent;
	settingsScreen: AppScreenContent;
	gameScreen: AppGameContent;
	restartModal: AppScreenContent;
	gameOverModal: AppScreenContent;
}

export interface AppScreenContent {
	title: string | string[];
	subtitle?: string | string[];
	cta1?: string;
	cta2?: string;
	errors?: {
		[key: string]: string;
	};
}

export interface AppGameContent {
	intelligenceLevel: [string, string, string];
	turnIndicator: [string, string];
	scoreBoard: [string, string, string];
}
