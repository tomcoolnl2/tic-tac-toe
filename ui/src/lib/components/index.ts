export { AppLogo } from './logo/logo';
export { Avatar } from './avatar/avatar';
export { Button } from './button/button';
export { Cell } from './cell/cell';
export { ChooseDifficulty } from './choose-difficulty/choose-difficulty';
export { GameOver } from './game-over/game-over';
export { Input } from './input/input';
export { Modal } from './modal/modal';
export { ReloadButton } from './reload-button/reload-button';
export { ScoreBoardItem } from './score-board-item/score-board-item';
export { Spinner } from './spinner/spinner';
export { SymbolChoice } from './symbol-choice/symbol-choice';
export { TurnIndicator } from './turn-indicator/turn-indicator';
