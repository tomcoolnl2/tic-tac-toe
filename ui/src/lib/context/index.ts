export * from './content/content';
export * from './auth/auth';
