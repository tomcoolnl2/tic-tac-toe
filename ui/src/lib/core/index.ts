export * from './canvas';
export * from './grid';
export * from './flex-box';
export * from './divider';
