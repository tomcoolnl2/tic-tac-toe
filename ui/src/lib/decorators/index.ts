export { withTheme } from './withTheme';
export type { ContainerProps } from './withContainer';
export { withContainer } from './withContainer';
