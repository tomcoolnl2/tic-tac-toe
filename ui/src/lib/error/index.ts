export * from './error-boundary';
export { AuthError } from '../context/auth/error';
