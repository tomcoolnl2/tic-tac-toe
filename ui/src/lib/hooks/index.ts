export * from './useAnimationFrame';
export * from './useBehaviorSubjectState';
export * from './useScreenOrientation';
export * from './useSettingsHandlers';
export * from './useInterfaceHandlers';
export * from './useTimer';
