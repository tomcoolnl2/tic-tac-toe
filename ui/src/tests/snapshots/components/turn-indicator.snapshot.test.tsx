import { render } from '@testing-library/react';
import { PlayerSymbol } from '@tic-tac-toe/model';
import { TurnIndicator } from '../../../lib/components';

describe('TurnIndicator component snapshot test', () => {
	it('should match the snapshot', () => {
		const { asFragment } = render(
			<TurnIndicator
				content={['Your turn', "CPU's Turn"]}
				playerSymbol={PlayerSymbol.X}
				currentPlayer={PlayerSymbol.X}
			/>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
