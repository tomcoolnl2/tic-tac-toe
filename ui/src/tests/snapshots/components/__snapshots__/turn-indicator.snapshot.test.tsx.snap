// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurnIndicator component snapshot test should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="button button-dark turn-indicator turn-indicator-x"
    disabled=""
  >
    <span
      class="avatar avatar-x avatar-size-s avatar-variant-light"
    />
    Your turn
  </button>
</DocumentFragment>
`;
